node/lib/util/status_util.js

```diff
@@ -387,11 +388,13 @@ exports.getRepoStatus = co.wrap(function *(repo, options) {
             const treeId = head.treeId();
             tree = yield NodeGit.Tree.lookup(repo, treeId);
         }
+        repo.submoduleCacheAll();
         const status = yield DiffUtil.getRepoStatus(repo,
                                                     tree,
                                                     options.paths,
                                                     options.ignoreIndex,
                                                     options.showAllUntracked);
+        repo.submoduleCacheClear();
```

Comment:
OK, there is at least one case (checkout) where we'd need to cache/uncache more than once. I'll refactor the other cases.